subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskDefinitionIntegrationSpec.groovy

```diff
@@ -207,6 +178,18 @@ class TaskDefinitionIntegrationSpec extends AbstractIntegrationSpec {
         'primitive' | '123'
     }
 
+    def "fails when constructor argument is wrong type"() {
+        given:
+        buildFile << CUSTOM_TASK_WITH_CONSTRUCTOR_ARGS
+        buildFile << "tasks.create('myTask', CustomTask, 123)"
```

Comment:
Wouldn't we need to add a `String` here, too, given that the constructor expects two arguments?